### ISSUE TITLE
Type sensors 'unknown' ? for croydon sensors and error in logs?

### ISSUE THREAD

**Issue by atv2016** [mampfes/hacs_waste_collection_schedule#1236]
So my food and glass sensors are due for today, yet the status of them is unknown. The calendar entity is turned to on though, and set to food waste.

If i do a fetch all sources as a service call i get this:

```
This error originated from a custom integration.

Logger: waste_collection_schedule.source_shell
Source: custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py:136 
Integration: waste_collection_schedule (documentation) 
First occurred: 19:59:27 (1 occurrences) 
Last logged: 19:59:27

fetch failed for source Croydon Council: Traceback (most recent call last): File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 134, in fetch entries = self._source.fetch() ^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/croydon_gov_uk.py", line 183, in fetch waste_date = pickup.find("div", {"class": "bin-collection-next"}).attrs["data-current_value"].strip() ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ AttributeError: 'NoneType' object has no attribute 'attrs'
```

This is my full configuration:
```yaml
waste_collection_schedule:
    sources:
    - name: croydon_gov_uk
      args:
        postcode: "CR0 5AA"
        houseID: "5 Inkheart Road"

# Waste Collection Croydon

  - platform: waste_collection_schedule
    name: food_waste
    types:
      - Food waste
  - platform: waste_collection_schedule
    name: general
    types:
      - General rubbish
  - platform: waste_collection_schedule
    name: paper
    types:
      - Paper and card recycling
  - platform: waste_collection_schedule
    name: glass
    types:
      - Glass, plastics, cans and cartons recycling
```

> edit by @5ila5 (added yaml formatting)

**Comment by dt215git** [comment]
The name/alias needs to be in the source, not the sensor.

```yaml
waste_collection_schedule:
  sources:
    - name: croydon_gov_uk
      args:
        postcode: "CR0 6EG"
        houseID: "48 Exeter Road"
      calendar_title: Croydon_Test
      customize:
        - type: "Food waste"
          alias: "food"
        - type: "General rubbish"
          alias: "rubbish"
        - type: "Paper and card recycling"
          alias: "paper"
        - type: "Glass, plastics, cans and cartons recycling"
          alias: "glass"
```

```yaml
- platform: waste_collection_schedule
  name: Croydon_Test_Food
  details_format: appointment_types
  add_days_to: true
  types:
    - food
- platform: waste_collection_schedule
  name: Croydon_Test_Rubbish
  details_format: appointment_types
  add_days_to: true
  types:
    - rubbish
- platform: waste_collection_schedule
  name: Croydon_Test_Paper
  details_format: appointment_types
  add_days_to: true
  types:
    - paper
- platform: waste_collection_schedule
  name: Croydon_Test_Glass
  details_format: appointment_types
  add_days_to: true
  types:
    - glass
```
![image](https://github.com/mampfes/hacs_waste_collection_schedule/assets/86194065/19d0e71a-e47b-455d-89c8-d72e637be38f)



However, the script is looking for a `bin-collection-next` to get the collection dates. It looks like that changes to `bin-collection-today` if the collection date is the same date the query is made. That would explain why you're seeing an "unknown", the script isn't looking for `bin-collection-today` entries. It'll need updating.

**Comment by atv2016** [comment]
Cool many thanks.

**Comment by atv2016** [comment]
@dt215git That makes sense. Who would be able to update the script?

**Comment by 5ila5** [comment]
Fix will be part of release 1.43.0 or is available now as master (git) version

**Comment by atv2016** [comment]
Nice! Thanks both!
________________________________
